pkg/snap/util/dqlite_test.go

```diff
@@ -84,3 +85,72 @@ Role: 0`,
 	})
 
 }
+
+func TestUpdateDqliteBindAddress(t *testing.T) {
+	s := &mock.Snap{
+		DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0`,
+		DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+	}
+
+	g := NewWithT(t)
+	findMatchingBindAddressMock := func(addr string) (string, error) {
+		return "10.10.10.10:19001", nil
+	}
+
+	// should fail if node already known
+	err := snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1", "127.0.0.1", findMatchingBindAddressMock)
+	g.Expect(err).To(MatchError("the joining node (127.0.0.1) is already known to dqlite"))
+
+	s = &mock.Snap{
+		DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0
+- Address: 10.10.10.10:19001
+  ID: 12345678987654321
+  Role: 0`,
+		DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+	}
+	// should not update bind address if cluster consists of multiple nodes
+	err = snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1:19001", "8.8.8.8", findMatchingBindAddressMock)
+	g.Expect(err).To(BeNil())
+	g.Expect(s.WriteDqliteUpdateYamlCalledWith).To(BeEmpty())
+
```

Comment:
I would recommend writing this as a sub-test instead of a comment, e.g.

```go
t.Run("MustNotUpdateIfMultipleNodes", func(t *testing.T) {
  ....
})
```

Reply:
Thanks for the suggestion - refactored it.